fvm/evm/emulator/state/collection.go

```diff
@@ -47,16 +47,16 @@ func NewCollectionProvider(
 // calling twice for the same collection might result in odd-behaviours
 // currently collection provider doesn't do any internal caching to protect aginast these cases
 func (cp *CollectionProvider) CollectionByID(collectionID []byte) (*Collection, error) {
-	storageID, err := atree.NewStorageIDFromRawBytes(collectionID)
+	slabID, err := atree.NewSlabIDFromRawBytes(collectionID)
 	if err != nil {
 		return nil, err
 	}
-	// sanity check the storage ID address
-	if storageID.Address != cp.rootAddr {
-		return nil, fmt.Errorf("root address mismatch %x != %x", storageID.Address, cp.rootAddr)
-	}
+	// TODO: sanity check the slab ID address
+	// if slabID.Address != cp.rootAddr {
+	// return nil, fmt.Errorf("root address mismatch %x != %x", storageID.Address, cp.rootAddr)
+	// }
```

Comment:
Is this a TODO for this PR?

Reply:
Good catch! :+1:  Added check back in commit [62c7780](https://github.com/onflow/flow-go/pull/5736/commits/62c77802b05e0dc93a3139c1bde3ed2461d4354b).